x-pack/legacy/plugins/siem/public/components/links/index.test.tsx

```diff
@@ -102,52 +108,235 @@ describe('Custom Links', () => {
   });
 
   describe('ReputationLink', () => {
-    test('it renders link text', () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={'192.0.2.0'}>{'Example Link'}</ReputationLink>
-      );
-      expect(wrapper.text()).toEqual('Example Link');
-    });
+    const mockCustomizedReputationLinks = [
+      { name: 'Link 1', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+      {
+        name: 'Link 2',
+        url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+      },
+      { name: 'Link 3', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+      {
+        name: 'Link 4',
+        url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+      },
+      { name: 'Link 5', url_template: 'https://www.virustotal.com/gui/search/{{ip}}' },
+      {
+        name: 'Link 6',
+        url_template: 'https://talosintelligence.com/reputation_center/lookup?search={{ip}}',
+      },
+    ];
+    const mockDefaultReputationLinks = mockCustomizedReputationLinks.slice(0, 2);
 
-    test('it renders correct href', () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={'192.0.2.0'}>{'Example Link'} </ReputationLink>
-      );
-      expect(wrapper.find('a').prop('href')).toEqual(
-        'https://www.talosintelligence.com/reputation_center/lookup?search=192.0.2.0'
-      );
-    });
+    describe('links property', () => {
+      beforeEach(() => {
+        (useUiSetting$ as jest.Mock).mockReset();
+        (useUiSetting$ as jest.Mock).mockReturnValue([mockDefaultReputationLinks]);
+      });
 
-    test("it encodes <script>alert('XSS')</script>", () => {
-      const wrapper = mountWithIntl(
-        <ReputationLink domain={"<script>alert('XSS')</script>"}>{'Example Link'}</ReputationLink>
-      );
-      expect(wrapper.find('a').prop('href')).toEqual(
-        "https://www.talosintelligence.com/reputation_center/lookup?search=%3Cscript%3Ealert('XSS')%3C%2Fscript%3E"
-      );
+      afterEach(() => {
+        (useUiSetting$ as jest.Mock).mockClear();
```

Comment:
Per the [Jest documentation for mockReset()](https://jestjs.io/docs/en/mock-function-api#mockfnmockreset), it:

> Does everything that mockFn.mockClear() does, and also removes any mocked return values or implementations.

Since the `beforeEach` above is already calling `mockReset` before providing a new implementation, consider removing the (seemingly) redundant `mockClear` here, and wherever this pattern is used.